pkg/kv/kvtest/store.go

```diff
@@ -369,3 +372,132 @@ func testScanPrefix(t *testing.T, ms MakeStore) {
 		t.Fatal("ScanPrefix entries didn't match:", diff)
 	}
 }
+
+func testDeleteWhileIterPrefix(t *testing.T, ms MakeStore) {
+	// iteration
+	//            deletion
+	testDeleteWhileIterPrefixSingleSequence(t, ms, "RRRRRRRDDDDDDD")
+	// iteration
+	//       deletion
+	testDeleteWhileIterPrefixSingleSequence(t, ms, "RRRDRDRDRDRDDD")
+	// iiiiiiiiiiteration
+	//     deletion
+	testDeleteWhileIterPrefixSingleSequence(t, ms, "RRRRRDRDRDRDRDRRRRR")
+	//   iteration
+	// deleeeeeeeetion
+	testDeleteWhileIterPrefixSingleSequence(t, ms, "DDDDRDRDRDRDDRDDDDD")
+	//     iteration
+	// deletion
+	testDeleteWhileIterPrefixSingleSequence(t, ms, "DDDDRDRDDRRDDDDD")
+	//           iteration
+	// deletion
+	testDeleteWhileIterPrefixSingleSequence(t, ms, "DDDDDRRRRR")
+}
+
+// Executing a sequence of (R)ead and (D)elete operations, according to the specified sequence
+func testDeleteWhileIterPrefixSingleSequence(t *testing.T, ms MakeStore, sequence string) {
```

Comment:
How do you make sure that all the R entries were actually read (non was skipped due to the cuncurrent deletion)? Or is it not part of the test?

Reply:
Originally, I had no intention to do so, but since you brought it up, I guess it is worth verifying
Done - thanks